modules/api_docs/php/api_docs.class.inc

```diff
@@ -0,0 +1,97 @@
+<?php declare(strict_types=1);
+/**
+ * PHP Version 7
+ *
+ * @category Documentation
+ * @package  Main
+ * @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+namespace LORIS\api_docs;
+use \Psr\Http\Message\ServerRequestInterface;
+use \Psr\Http\Message\ResponseInterface;
+
+/**
+ * This class list the schema files of this LORIS instance modules
+ * for the swagger-ui app to present. It is a public module but will only
+ * show specification (schemas) of module that the user can access.
+ *
+ * @category Documentation
+ * @package  Main
+ * @author   Xavier Lecours <xavier.lecours@mcin.ca>
+ * @license  http://www.gnu.org/licenses/gpl-3.0.txt GPLv3
+ * @link     https://www.github.com/aces/Loris-Trunk/
+ */
+
+class API_Docs extends \NDB_Page
+{
+    /**
+     * Determine whether the user has permission to view this page
+     *
+     * @param \User $user The user whose access is being checked
+     *
+     * @return bool whether the user has access
+     */
+    public function hasAccess(\User $user): bool
+    {
+        return $user->hasPermission('superuser');
```

Comment:
@xlecours @driusan any reason this is limited to superuser ?

Reply:
I don't know.

Reply:
Not sure but I remember Dave requesting this module not to be public. I might have added "superuser" as a temporary permission filter.